api/v1/reviewers/recommendations/ReviewerRecommendationController.php

```diff
@@ -0,0 +1,219 @@
+        $actionName = static::getRouteActionName($illuminateRequest);
+
+        $this->addPolicy(new UserRolesRequiredPolicy($request), true);
+        $this->addPolicy(new ContextAccessPolicy($request, $roleAssignments));
+
+        if (in_array($actionName, ['edit', 'updateStatus', 'delete'])) {
+            $this->addPolicy(
+                new RecommendationAccessPolicy(
+                    $request,
+                    static::getRequestedRoute($illuminateRequest)->parameter('reviewerRecommendationId')
+                )
+            );
+        }
+
+        return parent::authorize($request, $args, $roleAssignments);
+    }
+
+    /**
+     * @copydoc \PKP\core\PKPBaseController::getGroupRoutes()
+     */
+    public function getGroupRoutes(): void
+    {
+        Route::get('{reviewerRecommendationId}', $this->get(...))
+            ->name('reviewer.recommendations.get')
+            ->whereNumber(['reviewerRecommendationId']);
+
+        Route::get('', $this->getMany(...))
+            ->name('reviewer.recommendations.getMany');
+
+        Route::post('', $this->add(...))
+            ->name('reviewer.recommendations.add');
+
+        Route::put('{reviewerRecommendationId}', $this->edit(...))
+            ->name('reviewer.recommendations.edit')
+            ->whereNumber(['reviewerRecommendationId']);
+
+        Route::put('{reviewerRecommendationId}/status', $this->updateStatus(...))
+            ->name('reviewer.recommendations.edit.status')
+            ->whereNumber(['reviewerRecommendationId']);
+
+        Route::delete('{reviewerRecommendationId}', $this->delete(...))
+            ->name('reviewer.recommendations.delete')
+            ->whereNumber(['reviewerRecommendationId']);
+    }
+
+    /**
+     * Get specific recommendation response
+     */
+    public function get(Request $illuminateRequest): JsonResponse
+    {
+        $recommendation = ReviewerRecommendation::find($illuminateRequest->route('reviewerRecommendationId'));
+
+        return response()->json(
+            (new ReviewerRecommendationResource($recommendation))->toArray($illuminateRequest),
+            Response::HTTP_OK
+        );
+    }
+
+    /**
+     * Get all recommendations response
+     */
+    public function getMany(Request $illuminateRequest): JsonResponse
+    {
+        $recommendations = ReviewerRecommendation::query()
+            ->withContextId(Application::get()->getRequest()->getContext()->getId())
+            ->get();
+
+        return response()->json([
+            'items' => ReviewerRecommendationResource::collection($recommendations),
+            'itemMax' => $recommendations->count(),
+        ], Response::HTTP_OK);
+    }
+
+    /**
+     * Add new recommendation
+     */
+    public function add(AddReviewerRecommendation $illuminateRequest): JsonResponse
+    {
+        $validateds = $illuminateRequest->validated();
+
+        $recommendation = ReviewerRecommendation::create($validateds);
```

Comment:
Likewise, do we need to separately consider if the recommendation cannot be created and send a response accordingly?

Reply:
we can do that but will we get such as case ? After a successful validation along with proper guarded and cast data, we should always be able to create a new recommendation .

Reply:
In that case I don't think we need it, thanks!